### ISSUE TITLE
Spotify User ID Error

### ISSUE THREAD

**Issue by vodobaas1** [rnagabhyrava/plex-playlist-sync#45]
I'm sure I'm doing something wrong, but can't figure it out.
I have added my Spotify Client ID, Client Secret and User ID according to the Readme and Docker Setup and keep getting this error.  I know for a fact that I do have playlists.  My Deezer account is working fine, just Spotify.

INFO:root:Starting Spotify playlist sync
ERROR:root:Spotify User ID Error
ERROR:root:No spotify playlists found for given user

**Comment by rnagabhyrava** [comment]
This could be because there are no public playlists on your profile. Can you check and see if any playlists show up under https://open.spotify.com/user/{YOUR_USER_ID} without login? If so, Trying adding a playlist to your profile and rerun. You can do this by going into spotify app select any playlist and under options "Add to profile". Let me know if that worked.

**Comment by vodobaas1** [comment]
Thank you for the reply.  I went to [https://open.spotify.com/user/{YOUR_USER_ID}](https://open.spotify.com/user/%7BYOUR_USER_ID%7D) and have a few playlists there.  I tried to add a new one and rerun and still get the same issue.  Here is a screen shot of the playlists: (https://imgur.com/a/Q0AFOpV)

**Comment by rnagabhyrava** [comment]
Do they show up on your user page without sigin? Copy the profile url and paste it in incognito or other browser without login. The screenshot you posted in playlists in your account. You need to add them to your profile to make them public and for the program to catch them.

**Comment by vodobaas1** [comment]
Thanks, I tried it on incognito and it says "page not found."  Makes sense why it's not working with your system, but not sure why.

As a short term workaround I did import my spotify playlists into deezer and that's working for now until I figure out what's wrong on my spotify.

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rnagabhyrava** [comment]
Did you replace {YOUR_USER_ID} with your spotify user id? It's shouldn't show not found. It'll show your public playlists or empty.

**Comment by vodobaas1** [comment]
Yeah I must have mistyped my user ID.  It shows as empty.

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rnagabhyrava** [comment]
Then you did not follow my other comment on Adding playlists to your profile. Please add playlist playlists to your profile either through the app or website. I think web only allows you to add like 4 playlists to your profile so better use mobile app to add playlist to your profile.

Here is apost on how to do it: https://allthings.how/how-to-add-playlists-to-your-spotify-profile/

**Comment by vodobaas1** [comment]
Yeah, it still doesn't work.  I have some listed and it still doesn't pull.  https://open.spotify.com/user/122873654 has playlists added but still get the same error.

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rnagabhyrava** [comment]
Your playlists are public and are accessible by the program. I was able to grab your public playlists with no issue. Please check your config and restart the container. I was not able to reproduce the issue so cannot help you.

Here is a created playlist from your userid: https://imgur.com/a/ACQaSIf

This is docker config if it helps



> environment:
      - PLEX_URL=http://192.168.1.99:32400
      - PLEX_TOKEN=my_token
      - WRITE_MISSING_AS_CSV=1
      - SPOTIFY_CLIENT_ID=my_client_id
      - SPOTIFY_CLIENT_SECRET=my_client_secret
      - SPOTIFY_USER_ID=122873654 #I used your id to test

**Comment by vodobaas1** [comment]
I appreciate the help.  It was odd I checked all the settings I had to make sure my client ID, secret and user ID were correct, they all looked good and a compiled again (maybe my 10th time) and it is working now.  So not sure what the issue was recently since I didn’t change any variables, but I definitely didn’t have my playlists correct at first.  Thank you again for the help.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows
